### PR TITLE
Change the grad of abs to prevent relu from generating nan in fast_compile

### PR DESCRIPTION
**Pull request by nouiz**
fix gh-4353

### CODE REVIEW

File: theano/scalar/basic.py [hunk 1]
```diff
@@ -2150,6 +2150,8 @@ def grad(self, inputs, gout):
             else:
                 return [x.zeros_like()]
 
+        if x.dtype.startswith('float'):
+            return gz * sgn(x),
```
Comment (lamblin) [inline theano/scalar/basic.py]: That looks like a more stable formula, indeed.
Bud could you check for `x in float_dtypes` instead?

Comment (nouiz) [inline theano/scalar/basic.py]: done

**Comment by nouiz** [comment]
Thanks. I think it was quicker to do the small change then to describe them. So I did them here:

https://github.com/Theano/Theano/pull/4450

**Comment by nouiz** [comment]
Sorry, comment on the wrong branch.

**Comment by nouiz** [comment]
@lamblin this PR is ready to review.

**Pull request merged**